backend/epub/epub-document.c

```diff
@@ -1199,7 +1199,12 @@ setup_index_from_navfile(gchar *tocpath)
     GList *index = NULL;
     open_xml_document(tocpath);
     set_xml_root_node(NULL);
-    xmlNodePtr nav = xml_get_pointer_to_node((xmlChar*)"nav",(xmlChar*)"id",(xmlChar*)"toc");
+    xmlNodePtr nav = xml_get_pointer_to_node((xmlChar*)"nav",(xmlChar*)"type",(xmlChar*)"toc");
+
+    if (nav == NULL) {
+        return NULL;
```

Comment:
missing xml_free_doc

Reply:
Done - sorry, meant to comment earlier, I got distracted.